src/Chapter1.hs

```diff
@@ -537,7 +537,14 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid :: Int -> Int -> Int -> Int
+mid x y z 
+    | x == y = x
+    | y == z = y
+    | x == z = z
+    | max x (max y z) /= x && min z (min y x) /= x = x
+    | max x (max y z) /= y && min z (min y x) /= y = y
+    | max x (max y z) /= z && min z (min y x) /= z = z
```

Comment:
Wow, this is a really nice catch! I haven't thought about such a solution, very creative 🎨 

One side note. 
As we mentioned, the compiler in Haskell is very attentive to the exhaustive pattern-matching. And here it would warn you that `Pattern matching is not exhaustive`, as the guard have quite complicated logic, and the compiler won't be able to prove that it covers all the cases.
Because of that, you will need to add another guard (or write it instead of the final clause) – `| otherwise = z`, to tell the compiler, that your pattern matching is exhaustive 🙂